### ISSUE TITLE
Slayer plugin NPE

### ISSUE THREAD

**Issue by Nightfirecat** [runelite/runelite#14542]
Found in a support user's logs, appears to be due to `addCounter()` not guaranteeing that `counter` is set to a non-null value. (stemming from slayer xp gains when the plugin doesn't know the current task, for first-time RuneLite users for instance) Relevant line of code: https://github.com/runelite/runelite/blob/0da504769df58cb2f648d2fcc8048978c7997c5b/runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java#L645

```
2022-01-02 14:01:45 [Client] WARN  n.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.NullPointerException: null
	at net.runelite.client.plugins.slayer.SlayerPlugin.killed(SlayerPlugin.java:645)
	at net.runelite.client.plugins.slayer.SlayerPlugin.onStatChanged(SlayerPlugin.java:568)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:70)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:223)
	at net.runelite.client.callback.Hooks.post(Hooks.java:169)
	at client.mz(client.java:29351)
	at client.hf(client.java:7570)
	at client.gg(client.java:3685)
	at client.i(client.java:1843)
	at am.k(am.java:335)
	at am.run(am.java:314)
	at java.base/java.lang.Thread.run(Unknown Source)
```

**Comment by Doxterpepper** [comment]
I'm training slayer anyway so I'll take a look at this.

If this seems to be happening for players who haven't used runelite yet, could this occur if you start a slayer task through a different client and then continue the task through runelite?

**Comment by Adam-** [comment]
> I'm training slayer anyway so I'll take a look at this.
> 
> If this seems to be happening for players who haven't used runelite yet, could this occur if you start a slayer task through a different client and then continue the task through runelite?

You should be able to reproduce this by removing your task and then getting a slayer xp drop

**Comment by Adam-** [comment]
fixed in 1f6ae893d732cf3a022fe6d9662cd2a42ff2b217
